src/System.Globalization/tests/ISOWeekTests.cs

```diff
@@ -0,0 +1,215 @@
+        {
+            return s_YearData.Select(x => new object[] {x.Year, x.StartDate});
+        }
+
+        public static IEnumerable<object[]> GetYearEnd_TestData()
+        {
+            return s_YearData.Select(x => new object[] {x.Year, x.EndDate});
+        }
+
+        [Theory, MemberData(nameof(GetWeekOfYear_TestData))]
+        public static void GetWeekOfYear(DateTime date, int expected)
+        {
+            Assert.Equal(expected, ISOWeek.GetWeekOfYear(date));
+        }
+
+        [Theory, MemberData(nameof(GetYear_TestData))]
+        public static void GetYear(DateTime date, int expected)
+        {
+            Assert.Equal(expected, ISOWeek.GetYear(date));
+        }
+
+        [Theory, MemberData(nameof(ToDateTime_TestData))]
+        public static void ToDateTime(int year, int week, DayOfWeek dayOfWeek, DateTime expected)
+        {
+            Assert.Equal(expected, ISOWeek.ToDateTime(year, week, dayOfWeek));
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(10000)]
+        public static void ToDateTime_WithInvalidYear_Throws(int year)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => ISOWeek.ToDateTime(year, 1, DayOfWeek.Friday));
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(54)]
+        public static void ToDateTime_WithInvalidWeek_Throws(int week)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => ISOWeek.ToDateTime(2018, week, DayOfWeek.Friday));
+        }
+
+        [Theory]
+        [InlineData(-1)]
+        [InlineData(8)]
+        public static void ToDateTime_WithInvalidDayOfWeek_Throws(int dayOfWeek)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => ISOWeek.ToDateTime(2018, 1, (DayOfWeek)dayOfWeek));
+        }
+
+        [Theory, MemberData(nameof(WeeksInYear_TestData))]
+        public static void GetWeeksInYear(int year, bool isLong)
+        {
+            Assert.Equal(isLong ? 53 : 52, ISOWeek.GetWeeksInYear(year));
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(10000)]
+        public static void GetWeeksInYear_WithInvalidYear_Throws(int year)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => ISOWeek.GetWeeksInYear(year));
+        }
+
+        [Theory, MemberData(nameof(GetYearStart_TestData))]
+        public static void GetYearStart(int year, DateTime expected)
+        {
+            Assert.Equal(expected, ISOWeek.GetYearStart(year));
+        }
+
+        [Theory, MemberData(nameof(GetYearEnd_TestData))]
+        public static void GetYearEnd(int year, DateTime expected)
+        {
+            Assert.Equal(expected, ISOWeek.GetYearEnd(year));
+        }
+
+        private static DateTime ParseIsoDate(string date)
+        {
+            return DateTime.ParseExact(date, "yyyy-MM-dd", CultureInfo.InvariantCulture);
+        }
+
+        private struct DateData
```

Comment:
>DateData [](start = 23, length = 8)

why we need to use strings to parse it to initialize DateDate. can this be simplified by a constructor with the needed parameter? 


new DateDate(new DateTime(1, 1, 1), 1, 1, DayOfweek.Monday)? 

I know this is more typing but it is really simpler.

Reply:
We don't have to, I just thought it looked cleaner, but it should be quick to convert it 😊